DSCResources/cChocoPackageInstall/cChocoPackageInstall.psm1

```diff
@@ -77,13 +83,28 @@ function Set-TargetResource
         $AutoUpgrade = $false
     )
     Write-Verbose -Message 'Start Set-TargetResource'
+    $isVersionPresent = $PSBoundParameters.ContainsKey('Version')
+    $isMinimumVersionPresent = $PSBoundParameters.ContainsKey('MinimumVersion')
+    if ($isVersionPresent -and $isMinimumVersionPresent ) {
+        throw "Cannot specify 'Version' and 'MinimumVersion' in the same configuration"
+    }
 
     if (-Not (Test-ChocoInstalled)) {
         throw "cChocoPackageInstall requires Chocolatey to be installed, consider using cChocoInstaller with 'dependson' in dsc config"
     }
 
     $isInstalled = IsPackageInstalled -pName $Name
 
+    #Determine the correct package version to use get to desired state
+    if ($isVersionPresent -or $isMinimumVersionPresent) {
+        if ($isVersionPresent) {
+            $versionToInstall = $PSBoundParameters['Version']
+        }
+        else {
+            $versionToInstall = $PSBoundParameters['MinimumVersion']
+        }
+    }

```

Comment:
If you use Parameter Sets this can be simplified.

Reply:
Same as above. If I am missing something that makes parameter sets work correctly for DSC I will happily implement it.